File: 크레인 인형뽑기 게임.c
#include <stdio.h>

int		solution(int board[5][5], int *moves)
{
	int stack[1000] = {0, };
	int answer = 0;
	int k = 0;
	int move = 0;
	for (int i = 0; moves[i] != 0; i++)
		for (int j = 0; j < 5; j++)
			if (board[j][moves[i] - 1] > 0)
			{
				stack[k] = board[j][moves[i] - 1];
				board[j][moves[i] - 1] = 0;
				if (k > 0 && stack[k] == stack[k - 1])
				{
					answer++;
					stack[k] = 0;
				}
				printf("stack : %d, answer : %d\n",stack[k], answer);
				k++;
				break;
			}
	printf("\n");
	return (answer);
}

int		main(void)
{
	int board[5][5] = {
		{0,0,0,0,0},
		{0,0,1,0,3},
		{0,2,5,0,1},
		{4,2,4,4,2},
		{3,5,1,3,1}};
	int	moves[8] = {1,5,3,5,1,2,1,4};
	int answer = solution(board, moves);
	printf("%d", answer);
	return (0);
}

File: 크레인 인형뽑기 게임.py
def solution(board, moves):
    ans = 0
    stack = list()
    k = 0
    for i in range(len(moves)):
        for j in range(len(board[0])):
            if board[j][moves[i] - 1] > 0:
                stack.append(board[j][moves[i] - 1])
                board[j][moves[i] - 1] = 0
                if (k > 0 and stack[k] == stack[k - 1]):
                    ans += 2
                    stack.pop()
                    stack.pop()
                    k -= 2
                k += 1
                break
    return ans

board = [[0,0,0,0,0,0],
         [0,0,1,0,3,0],
         [0,2,5,0,1,3],
         [4,2,4,4,2,1],
         [3,5,1,3,1,4],
         [1,1,1,1,1,1]]
moves = [1,5,6,3,5,1,6,2,1,4]
answer = 0

answer = solution(board, moves)
print(answer)
